service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java

```
@@ -767,6 +767,18 @@ Object[] getListAsArray(List<?> attVal, DataTypeMapping typeMapping) {
           .map(Boolean::parseBoolean)
           .toList()
           .toArray(new Boolean[0]);
+      case ARRAY_OF_JSON -> attVal.stream()
+          .map(
+              el -> {
+                try {
+                  return objectMapper.writeValueAsString(el);
+                } catch (JsonProcessingException e) {
+                  LOGGER.error("Could not serialize array element to json string", e);
+                  throw new RuntimeException(e);
```

thinking out loud … if a user ever ended up with invalid data that couldn't be serialized, this would prevent them from seeing anything at all, any time the invalid row was in their query. Should this return `el.toString` instead, or even return the string "ERROR" here? I'm waffling on this, since if we swallow the error, an end user might never notice the problem and run Cromwell analyses on this …

---

I'm leaning towards fail loudly here. I think this shouldn't ever happen, since the data must have been parsed as JSON when uploaded.